SANC 70: Remake login page

jason-duong4509 opened the pull request:
Changed the sign in page to better reflect the UI design in the figma. See the images below 

<img width="959" height="497" alt="image" src="https://github.com/user-attachments/assets/ded96990-c7a4-440e-8dea-59d2fd331cb0" />

Desktop view

<img width="201" height="415" alt="image" src="https://github.com/user-attachments/assets/9de241e1-b30a-420d-b50c-2828d6d30b7f" />

Mobile view

Note that the "Contact an administrator" button requires a real email for it to function properly. It currently has a placeholder email address for now

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Redesigned login page with skyline background, centered layout, and logo header
  * Added password input to the login form
  * Added account prompt with "Contact an administrator" mailto link
  * Updated submit button label to "Log in" and restyled forgot-password link

* **Style**
  * New styling for login page elements: title, form card, spacing, responsive height cap, and logo row

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jason-duong4509 committed:
SANC 70: Finished Ticket

promatty commented on src/app/login/page.tsx:
```diff
@@ -51,41 +54,61 @@ export default function LoginPage() {
-              required
-              {...form.getInputProps("email")}
-            />
+          <Title order={1} size="22px" ta="center" pt="20px" pb="5px" mb="md">
```
can we add these inline styles for the Title into the scss module for better maintainability

jason-duong4509 replied:
Done!

promatty commented on src/app/login/page.tsx:
```diff
@@ -51,41 +54,61 @@ export default function LoginPage() {
+              </Button>
+
+              <Text ta="center" lh="1">
+                <Text component="span" size="13px" c="#228BE6">
```
isnt this the same as c="blue"?

jason-duong4509 replied:
Yes. I wrote it in hexadecimal. I did, however, change the color from the bright blue into cobalt blue since I made the mistake of not using the pre-defined colors in `\src\styles\_variables.scss`

<img width="288" height="369" alt="image" src="https://github.com/user-attachments/assets/1b1fa09b-22fb-4cc6-a74e-31dcf093ac2b" />

I think this is a better fix but let me know if there are any suggestions for this

promatty commented on src/app/login/page.tsx:
```diff
@@ -51,41 +54,61 @@ export default function LoginPage() {
+                  Don't have an account?{" "}
+                </Text>
+
+                <Text component="span" size="13px" c="dimmed" td="underline">
```
looks like we use size 13px many times. consider turning it into a constant variable or define it in a scss variable

jason-duong4509 replied:
Done! It's a SCSS variable now

promatty commented on src/app/login/page.tsx:
```diff
@@ -51,41 +54,61 @@ export default function LoginPage() {
+        <Paper shadow="0px 0px 10px rgba(0, 0, 0, 0.25)" p="xl" radius="md">
+          <div className={ui.logo}>
+            <SalvationLogo></SalvationLogo>
+            <AlbertaLogo></AlbertaLogo>
```
yeah self close

jason-duong4509 replied:
Done!

promatty commented on src/app/login/page.tsx:
```diff
@@ -51,41 +54,61 @@ export default function LoginPage() {
-          Sign In
-        </Title>
+    <div className={ui.background}>
+      <div className={styles.container}>
```
is this styles.container div redundant if youre wrapping the component in a Paper?

jason-duong4509 replied:
No. Removing that will cause the UI to expand outwards. I kept it in the branch as a result

promatty reviewed (changes requested):
in general, move reused styles into a scss variable. also try not to use magic numbers (hard-coded numbers) where possible, they should instead be variables.

jason-duong4509 committed:
SANC-70: Added Fixes to Address Feedback

- Moved all styles from the .tsx file into the .scss file for better
  maintainability
- Removed a redundant HTML element
- Closed two HTML elements since they weren't syntactically proper
- Changed the text color from blue to the pre-defined cobalt blue to
  match the figma

promatty commented on src/app/login/Login.module.scss:
```diff
@@ -0,0 +1,66 @@
+  text-decoration: underline;
+}
+
+.forgotPass {
```
there is complete overlap with .dontHaveAcc. make a new style for links, and both will extend it

jason-duong4509 replied:
I am not sure which styles you were referring to. I assumed that you meant `.forgotPass` and `.dontHaveAcc`

I changed `.dontHaveAcc` to `.links` (naming) and made `.forgotPass` and `.contactAdmin` extend it since both require additional styles that do not overlap

Let me know if there's something else I missed

promatty commented on src/app/login/page.tsx:
```diff
@@ -51,41 +54,59 @@ export default function LoginPage() {
+                  {/*TODO: CHANGE CONTACT EMAIL TO SALVATION ARMY ONE ONCE KNOWN V*/}
+                  <Anchor href="mailto:changeEmail@AtSomePoint.com" className={ui.contactAdmin}>
+                    Contact an administrator
+                  </Anchor>
```
duplicated ui.contactAdmin style

jason-duong4509 replied:
Fixed!

promatty commented on src/app/login/Login.module.scss:
```diff
@@ -0,0 +1,66 @@
+  color: $color-cobalt-blue;
+  font-size: $text-size;
+  padding-top: 3.5px;
+  padding-bottom: 4px;
```
actually, does forgotPass even need this padding? why cant we just make it the same as dontHaveAcc

jason-duong4509 replied:
Not really. The padding does provide some use and I only included it there because the conversion process from writing the styles in-line to a .scss file resulted in that padding being added

<img width="478" height="494" alt="image" src="https://github.com/user-attachments/assets/043c6347-d2e2-421f-a510-7a2908b29015" />

This (A) is the UI with the padding

<img width="479" height="497" alt="image" src="https://github.com/user-attachments/assets/3a6be644-4b58-480b-be50-3c6077dd1088" />

This (B) is the UI without the padding

The difference is very marginal. For simplicity, I removed it. Let me know if that works with you

jason-duong4509 committed:
SANC-70: More Fixes to Address Feedback

- Removed an unneeded Text element
- Cleaned up style definitions for .forgotPass by having it extend overlapping styles from another element
- Removed unneeded styling for .forgotPass
- Added styling to .forgotPass to account for an error where the user could click on the link from the right side of the form

promatty reviewed (approved):
lgtm
